From foo@baz Wed Aug 12 09:28:09 PDT 2009
Date: Wed, 12 Aug 2009 09:28:09 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: rtl8192e: backport net_device_ops conversion

2.6.27 does not have struct net_device_ops, so backport the changes
needed to get things to work here.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192e/ieee80211/ieee80211_module.c |    2 --
 drivers/staging/rtl8192e/r8192E_core.c                |    6 ++++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/rtl8192e/ieee80211/ieee80211_module.c
+++ b/drivers/staging/rtl8192e/ieee80211/ieee80211_module.c
@@ -118,9 +118,7 @@ struct net_device *alloc_ieee80211(int s
 #else
 	ieee = (struct ieee80211_device *)dev->priv;
 #endif
-#if 0
 	dev->hard_start_xmit = ieee80211_xmit;
-#endif
 
 	memset(ieee, 0, sizeof(struct ieee80211_device)+sizeof_priv);
 	ieee->dev = dev;
--- a/drivers/staging/rtl8192e/r8192E_core.c
+++ b/drivers/staging/rtl8192e/r8192E_core.c
@@ -6231,6 +6231,7 @@ void rtl8192_irq_rx_tasklet(struct r8192
        write_nic_dword(priv->ieee80211->dev, INTA_MASK,read_nic_dword(priv->ieee80211->dev, INTA_MASK) | IMR_RDU);
 }
 
+#if 0
 static const struct net_device_ops rtl8192_netdev_ops = {
 	.ndo_open =			rtl8192_open,
 	.ndo_stop =			rtl8192_close,
@@ -6241,6 +6242,7 @@ static const struct net_device_ops rtl81
 	.ndo_set_mac_address =		r8192_set_mac_adr,
 	.ndo_start_xmit = 		ieee80211_xmit,
 };
+#endif
 
 /****************************************************************************
      ---------------------------- PCI_STUFF---------------------------
@@ -6362,8 +6364,9 @@ static int __devinit rtl8192_pci_probe(s
 	dev->irq = pdev->irq;
 	priv->irq = 0;
 
-	dev->netdev_ops = &rtl8192_netdev_ops;
 #if 0
+	dev->netdev_ops = &rtl8192_netdev_ops;
+#endif
 	dev->open = rtl8192_open;
 	dev->stop = rtl8192_close;
 	//dev->hard_start_xmit = rtl8192_8023_hard_start_xmit;
@@ -6372,7 +6375,6 @@ static int __devinit rtl8192_pci_probe(s
 	dev->do_ioctl = rtl8192_ioctl;
 	dev->set_multicast_list = r8192_set_multicast;
 	dev->set_mac_address = r8192_set_mac_adr;
-#endif
 
          //DMESG("Oops: i'm coming\n");
 #if WIRELESS_EXT >= 12
